third_party/blink/renderer/core/inspector/inspector_dom_debugger_agent.cc

```diff
@@ -724,27 +727,51 @@ void InspectorDOMDebuggerAgent::ScriptExecutionBlockedByCSP(
   PauseOnNativeEventIfNeeded(std::move(event_data), true);
 }
 
+// Call `ReplayOnEvent` for the given event *before* it happened.
+static void ReplayNotifyBeforeEvent(const String& eventName,
+                              EventTarget* eventTarget = nullptr,
+                              bool isCallback = false);
+
+// Call `ReplayOnEvent` for the given event *after* it happened.
+static void ReplayNotifyAfterEvent(const String& eventName,
+                            EventTarget* eventTarget = nullptr,
+                            bool isCallback = false);
+
 void InspectorDOMDebuggerAgent::Will(const probe::ExecuteScript& probe) {
+  ReplayNotifyBeforeEvent("scriptFirstStatement");
   AllowNativeBreakpoint("scriptFirstStatement", nullptr, false);
 }
 
 void InspectorDOMDebuggerAgent::Did(const probe::ExecuteScript& probe) {
+  ReplayNotifyAfterEvent("scriptFirstStatement");
   CancelNativeBreakpoint();
 }
 
 void InspectorDOMDebuggerAgent::Will(const probe::UserCallback& probe) {
   String name = probe.name ? String(probe.name) : probe.atomic_name;
+
   if (probe.event_target) {
     Node* node = probe.event_target->ToNode();
     String target_name =
         node ? node->nodeName() : probe.event_target->InterfaceName();
+    ReplayNotifyBeforeEvent(name, probe.event_target, false);
     AllowNativeBreakpoint(name, &target_name, false);
     return;
   }
+  ReplayNotifyBeforeEvent(name, nullptr, true);
   AllowNativeBreakpoint(name + ".callback", nullptr, false);
 }
 
 void InspectorDOMDebuggerAgent::Did(const probe::UserCallback& probe) {
+  String name = probe.name ? String(probe.name) : probe.atomic_name;
+
+  if (probe.event_target) {
+    ReplayNotifyAfterEvent(name, probe.event_target, false);
+  }
+  else {
```

Comment:
formatting

Reply:
Sigh... Of all the formatting rules, same-line `else` is probably the one I dislike the most